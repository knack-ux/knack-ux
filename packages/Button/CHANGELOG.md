# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.1](https://github.com/chrispcode/knack/compare/@chrispcode/knack-button@0.3.0...@chrispcode/knack-button@0.3.1) (2019-09-26)


### Bug Fixes

* **button:** add styled.tsx for styled-components ([af93088](https://github.com/chrispcode/knack/commit/af93088)), closes [#1](https://github.com/chrispcode/knack/issues/1)





# [0.3.0](https://github.com/chrispcode/knack/compare/@chrispcode/knack-button@0.2.0...@chrispcode/knack-button@0.3.0) (2019-09-22)


### Features

* **button:** add wrapper component and add docs ([c56dfa3](https://github.com/chrispcode/knack/commit/c56dfa3))





# 0.2.0 (2019-09-22)


### Features

* **initial:** initial commit ([0d606fa](https://github.com/chrispcode/knack/commit/0d606fa))
* **storybook, github actions:** add storybook and gh actions ([7ca482b](https://github.com/chrispcode/knack/commit/7ca482b))
