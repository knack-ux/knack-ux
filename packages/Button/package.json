{
  "name": "@knack-ux/button",
  "version": "0.10.1",
  "description": "Button component for the Knack UI Project",
  "main": "lib/index.js",
  "repository": {
    "type": "git",
    "url": "ssh://git@github.com/knack-ux/knack-ux.git",
    "directory": "@knack-ux/button"
  },
  "files": [
    "lib/"
  ],
  "author": "Kristiyan Serafimov (chrispcode)",
  "license": "MIT",
  "scripts": {
    "start": "webpack --watch --config=../../webpack/webpack.dev.js",
    "build": "webpack --config=../../webpack/webpack.prod.js",
    "build:doc": "webpack --config=../../webpack/webpack.dev.js",
    "clean": "rm -rf lib"
  },
  "publishConfig": {
    "access": "public"
  },
  "peerDependencies": {
    "react": "^16.8.4",
    "styled-components": "^4.1.3",
    "styled-system": "^5.1.2"
  },
  "dependencies": {
    "@knack-ux/theme": "^0.7.2",
    "polished": "^3.4.2"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^5.0.2",
    "@testing-library/react": "^9.4.0"
  },
  "gitHead": "259d59c20c130d889bb07835c16f1b8bda3dff43"
}
