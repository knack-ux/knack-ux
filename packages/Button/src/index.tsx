import React, { forwardRef, Ref, ComponentPropsWithRef } from 'react';
import { LayoutProps, SpaceProps } from 'styled-system';
import Icon from '@knack-ux/icon';

import { ButtonBase } from './styled';

type ReactButton = ComponentPropsWithRef<'button'>

export interface Props extends ReactButton, SpaceProps, LayoutProps {
  appearance?: 'default' | 'primary' | 'minimal'
  intent?: string
  icon?: string
  iconRight?: string
  circular?: boolean
  loading?: boolean
}

export const Button = forwardRef<HTMLButtonElement, Props>((
  {
    appearance = 'default',
    intent = 'info',
    icon,
    iconRight,
    children,
    disabled,
    onClick,
    ...props
  },
  ref
) => (
  <ButtonBase
    intent={intent}
    appearance={appearance}
    aria-disabled={disabled}
    type="button"
    ref={ref}
    onClick={!disabled ? onClick : () => {}}
    {...props}
  >
    {icon && (
      <Icon
        icon={icon}
        mr={children ? '8px' : '0'}
      />
    )}
    {children}
    {iconRight && (
      <Icon
        icon={iconRight}
        ml="8px"
      />
    )}
  </ButtonBase>
));


export default Button;
