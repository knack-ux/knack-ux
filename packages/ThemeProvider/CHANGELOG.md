# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.7.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.6.0...@knack-ux/theme@0.7.0) (2019-12-21)


### Features

* **badge:** create badge component ([36bd405](https://github.com/knack-ux/knack-ux/commit/36bd405))





# [0.6.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.5.0...@knack-ux/theme@0.6.0) (2019-12-14)


### Features

* **icon, avatar:** create components ([0901f92](https://github.com/knack-ux/knack-ux/commit/0901f92))





# [0.5.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.4.0...@knack-ux/theme@0.5.0) (2019-12-09)


### Features

* **packages:** remove most of them clean ([06aa5a9](https://github.com/knack-ux/knack-ux/commit/06aa5a9))





# [0.4.0](https://github.com/knack-ux/knack-ux/compare/@knack-ux/theme@0.3.0...@knack-ux/theme@0.4.0) (2019-11-28)


### Features

* **package.json:** add repository fields ([36c89ad](https://github.com/knack-ux/knack-ux/commit/36c89ad))





# 0.3.0 (2019-11-26)


### Bug Fixes

* **input:** typings ([2b922ad](https://github.com/chrispcode/knack/commit/2b922ad))


### Features

* **initial:** initial commit ([0d606fa](https://github.com/chrispcode/knack/commit/0d606fa))
* **many:** rewrite some components ([800dab3](https://github.com/chrispcode/knack/commit/800dab3))
* **storybook, github actions:** add storybook and gh actions ([7ca482b](https://github.com/chrispcode/knack/commit/7ca482b))





# 0.2.0 (2019-09-22)


### Features

* **initial:** initial commit ([0d606fa](https://github.com/chrispcode/knack/commit/0d606fa))
* **storybook, github actions:** add storybook and gh actions ([7ca482b](https://github.com/chrispcode/knack/commit/7ca482b))
