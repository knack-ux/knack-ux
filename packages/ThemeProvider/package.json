{
  "name": "@knack-ux/theme",
  "version": "0.2.0",
  "description": "Theme provider for the Knack UI Project",
  "main": "lib/index.js",
  "repository": "https://github.com/chrispcode/knack",
  "author": "Kristiyan Serafimov (chrispcode)",
  "license": "MIT",
  "scripts": {
    "start": "webpack --watch --config=../../webpack/webpack.dev.js",
    "build": "webpack --config=../../webpack/webpack.prod.js",
    "build:doc": "webpack --config=../../webpack/webpack.dev.js",
    "clean": "rm -rf lib"
  },
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "peerDependencies": {
    "react": "^16.8.4",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "lodash.get": "^4.4.2"
  }
}
