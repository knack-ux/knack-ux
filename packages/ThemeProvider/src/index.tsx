import React from 'react';
import { ThemeProvider as Base, DefaultTheme } from 'styled-components';
import _get from 'lodash.get';


// and extend them!


export const defaultTheme = {
  a11y: {
    default: '#0052CC',
    info: '#0052CC',
    success: '#00855f',
    warning: '#B15f00',
    danger: '#D63B00'
  },
  colors: {
    default: '#0052CC',
    info: '#0052CC',
    success: '#3DA68C',
    warning: '#FF8B00',
    danger: '#FF5630'
  },
  focus: '#CED6E0'
};


declare module 'styled-components' {
  export interface DefaultTheme {
    a11y: {
      default: string
      success: string
      warning: string
      danger: string
    },
    colors: {
      default: string
      success: string
      warning: string
      danger: string
    },
    focus: string
  }
}

export function ThemeProvider({
  children
}: any) {
  return (
    <Base theme={defaultTheme}>
      {children}
    </Base>
  );
}

export default ThemeProvider;
