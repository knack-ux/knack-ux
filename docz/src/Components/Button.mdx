---
name: Button
route: /button
menu: Components
---

import { useState } from 'react';
import { Playground, Props } from 'docz';
import { Primary, Secondary } from '@knack/button';
import { version } from '@knack/button/package.json'

# Button
<span>Current version: {version}</span>

```js

import Button from '@knack/button';

```
### Props - `All html button props`

## Primary (Default)
<Playground>
  <Primary>Continue</Primary>
</Playground>

## Secondary
<Playground>
  <Secondary>Press</Secondary>
</Playground>

