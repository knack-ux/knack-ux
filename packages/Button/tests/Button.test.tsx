import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import ThemeProvider from '../../ThemeProvider/src';
import Button from '../src';

function Root({ children }) {
  return (
    <ThemeProvider>
      {children}
    </ThemeProvider>
  );
}

describe('Button', () => {
  it('should add an icon if icon prop passed', () => {
    const { getByText } = render(
      <Root>
        <Button icon="error">
          Continue
        </Button>
      </Root>
    );

    const iconElement = getByText('error');

    expect(iconElement).toBeTruthy();
    expect(iconElement).toHaveStyle('margin-right: 8px;');
  });

  it('should add an icon if icon prop passed', () => {
    const { getByText } = render(
      <Root>
        <Button icon="error" />
      </Root>
    );

    const iconElement = getByText('error');

    expect(iconElement).toHaveStyle('margin-right: 0');
  });

  it('should match disabled snapshot', () => {
    const { container } = render(
      <Root>
        <Button disabled>
          Continue
        </Button>
      </Root>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <Root>
        <Button>
          Continue
        </Button>
      </Root>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
