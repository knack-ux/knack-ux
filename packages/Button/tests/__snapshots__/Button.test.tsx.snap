// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match disabled snapshot 1`] = `
<button
  aria-disabled="true"
  class="sc-AxjAm CImRz"
  type="button"
>
  Continue
</button>
`;

exports[`Button should match snapshot 1`] = `
<button
  class="sc-AxjAm jwRqPu"
  type="button"
>
  Continue
</button>
`;
